benchopt/cli/main.py

```diff
@@ -183,6 +185,12 @@ def run(config_file=None, **kwargs):
         deprecated_objective_filters, old_objective_filters
     ) = _get_run_args(kwargs, config)
 
+    try:
+        timeout = int(timeout)
+    except ValueError:  # already under string format
+        import pandas as pd
+        timeout = pd.to_timedelta(timeout).total_seconds()
```

Comment:
This won't work in an environment, as this function returns a float, which gets casted as a string that is not properly process here. In order to make this more robust, I would say either we try to cast floats in line 189 `int(float(timeout))`, we cast the output of `192` to `int`, or we move this processing in the `run_benchmark` (but maybe it is better to check validity before launching computations in another env).

Reply:
int(float(timeout)) seems the best to me!

Reply:
(because it also supports the edge case `--timeout 100.` passed from command line)

Thanks for the feedback, all addressed

Reply:
agreed, it seemed the best.